pennylane/ops/qubit/state_preparation.py

```diff
@@ -199,14 +200,27 @@ def state_vector(self, wire_order=None):
             raise WireError("Custom wire_order must contain all QubitStateVector wires")
 
         num_total_wires = len(wire_order)
-        indices = tuple([slice(None)] * num_op_wires + [0] * (num_total_wires - num_op_wires))
-        ket = np.zeros((2,) * num_total_wires, dtype=np.complex128)
+        indices = tuple(
+            [Ellipsis] + [slice(None)] * num_op_wires + [0] * (num_total_wires - num_op_wires)
+        )
+        ket_shape = [2] * num_total_wires
+        if self.batch_size:
+            # Add broadcasted dimension to the shape of the state vector
+            ket_shape.insert(0, self.batch_size)
+
+        ket = np.zeros(ket_shape, dtype=np.complex128)
         ket[indices] = op_vector
 
         # unless wire_order is [*self.wires, *rest_of_wire_order], need to rearrange
         if self.wires != wire_order[:num_op_wires]:
             current_order = self.wires + list(Wires.unique_wires([wire_order, self.wires]))
             desired_order = [current_order.index(w) for w in wire_order]
+            if self.batch_size:
+                # If the operation is broadcasted, the desired order must include the batch dimension
+                # as the first dimension.
+                desired_order = [d + 1 for d in desired_order]
+                desired_order.insert(0, 0)
```

Comment:
Sometimes we use tuples, sometimes lists.
Sometimes we do `[first_val] + list_of_rest_vals`, sometimes `list_of_rest_vals.insert(0, first_val)`.

Do we have criteria for when we should use which? Does it matter? I feel like these fluctuations are a bit funny (no too specific to this PR, just noticed it quite clearly here)

Reply:
I think we have no standard or preferred way. I don't feel very strongly about this because it's always scoped to one or two lines and I'm assuming they're equivalent performance-wise, but I'm happy to set a standard. Consistency is definitely always nice within a project, so if it's sticking out to you then maybe let's pick one! How about `[your_prefix] + [existing_list]`? Allows for generally inserting more than one value to the front, plus this could be written in one line, and I love reducing LOC :)

Reply:
Definitely no strong preference either. I just was wondering whether we should indeed unify :) 
I timed the list addition vs insertion and they are indeed very similar in performance.